content/en/docs/Advanced/sys-extensions.md

```diff
@@ -0,0 +1,156 @@
+            ✗ confext for portable service
+            ✗ confext for initrd
+
+RW DESIGNATOR      PARTITION UUID                       PARTITION LABEL        FSTYPE                AR>
+ro root            4afae1e5-c73c-2f5a-acdc-3655ed91d4e0 root-x86-64            erofs                 x8>
+ro root-verity     abea5f2f-214d-4d9f-83f8-ee69ca7614ba root-x86-64-verity     DM_verity_hash        x8>
+ro root-verity-sig bdb3ee65-ed86-480c-a750-93015254f1a7 root-x86-64-verity-sig verity_hash_signature x8>
+```
+
+
+### Adding system extensions to build medium
+
+Check the `Bundling system extensions during the installable medium build` section in the [Trusted Boot Installations]({{%relref "/docs/Installation/trustedboot" %}}) for more information on how to add system extensions to the build medium.
+
+
+### Adding system extensions to the installed system
+
+Its possible to add system extension s once the system is installed or as part of the installation process without having to bundle them directly on the install media. This is done by copying the sysexts directly into the EFI partition in the proper place.
+
+The paths to copy the sysexts are:
+- `EFI/kairos/active.efi.extra.d/` for the sysexts that will be loaded when choosing the active boot entry.
+- `EFI/kairos/passive.efi.extra.d/` for the sysexts that will be loaded when choosing the passive boot entry.
+
+For any other entries, the sysexts can be added to the `EFI/kairos/$EFI_ENTRY_NAME.extra.d/` directory directly. So if we had a custom entry in which the EFI file is called `customentry.efi` the path to add the sysexts would be `EFI/kairos/customentry.efi.extra.d/`.
+
+This can be done either manually once installation is finished, by mounting the EFI partition, creating the dirs and copying the sysexts, or by adding a step to the install configuration that will do this automatically (`after-install` stage).
+
+We recommend bundling the sysexts during the build process, but this is an alternative for those cases where the sysexts are not known at build time or need to be added later to an existing system.
+
+This is also a good choice for testing sysexts before bundling them into the install media as they can be added to just one entry in the system and tested without affecting other entries.
+
+
+### Boot workflow
+
+During boot, systemd-stub will copy and measure the system extensions from the proper directory according to the loaded EFI file (`/EFI/kairos/$EFI_FILE.efi.extra.d/`) and copy them into the initrd `/.extra/sysext` directory.
+Immucore then will extract the public keys from the firmware and save them under `/run/verity.d` so the systemd extensions can be veried against those.
+Immucore will then verify those sysexts to see if they are signed and verity protected. If they are, they will be copied under `/run/extensions` and during the `initramfs` yip stage the `systemd-sysext` service will be run and they will be mounted in their proper directories.
```

Comment:
So it's immucore verifying those? Does systemd itself do any verification?

Reply:
its actually systemd-dissect confirming that the sysext its ok and signed+verity